autorest/preparer.go

```diff
@@ -80,6 +80,16 @@ func WithNothing() PrepareDecorator {
 	}
 }
 
+// Nest returns a PrepareDecorator that nests the prepare decorators together.
+func Nest(pds ...PrepareDecorator) PrepareDecorator {
```

Comment:
I don't think `Nest` is the best name here. Maybe `JoinPrepareDecorators` (as Go does not support method overloading, this wouldn't be very cool in case we wanted to also implement a similar func for ResponseDecorators).